Index: wp-includes/general-template.php
===================================================================
--- wp-includes/general-template.php	(revision 37156)
+++ wp-includes/general-template.php	(working copy)
@@ -1668,8 +1668,14 @@
 		$order = 'DESC';
 	}
 
-	// this is what will separate dates on weekly archive links
-	$archive_week_separator = '&#8211;';
+	/**
+	 * Filter what will separate dates on weekly archive links.
+	 *
+	 * @since 4.5
+	 *
+	 * @param string '-' (&#8211;) used to split the atart and end for the week.
+	 */
+	$archive_week_separator =  apply_filters( 'getarchives_week_separator', '&#8211;' );
 
 	$sql_where = $wpdb->prepare( "WHERE post_type = %s AND post_status = 'publish'", $r['post_type'] );
 
@@ -1790,7 +1796,12 @@
 					$arc_week       = get_weekstartend( $result->yyyymmdd, get_option( 'start_of_week' ) );
 					$arc_week_start = date_i18n( get_option( 'date_format' ), $arc_week['start'] );
 					$arc_week_end   = date_i18n( get_option( 'date_format' ), $arc_week['end'] );
-					$url            = sprintf( '%1$s/%2$s%3$sm%4$s%5$s%6$sw%7$s%8$d', home_url(), '', '?', '=', $arc_year, '&amp;', '=', $result->week );
+
+					$url = add_query_arg( array(
+						'm' => $arc_year,
+						'w' => $result->week,
+					), home_url() );
+
 					if ( 'post' !== $r['post_type'] ) {
 						$url = add_query_arg( 'post_type', $r['post_type'], $url );
 					}
