Index: wp-login.php
===================================================================
--- wp-login.php	(revision 36252)
+++ wp-login.php	(working copy)
@@ -678,8 +678,10 @@
 		 *
 		 * @param string $sign_up_url The sign up URL.
 		 */
-		wp_redirect( apply_filters( 'wp_signup_location', network_site_url( 'wp-signup.php' ) ) );
-		exit;
+		if ( wp_redirect( apply_filters( 'wp_signup_location', network_site_url( 'wp-signup.php' ) ) ) ){
+			exit;
+		};
+
 	}
 
 	if ( !get_option('users_can_register') ) {
